### PR TITLE
Consistently use 3D, 2D, 1D for dimensionality

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

As documented in the [docstring guide](https://matplotlib.org/devdocs/devel/documenting_mpl.html#parameter-type-descriptions).

### CODE REVIEW

**Review by dopplershift — APPROVED** [review]

**Comment by story645** [comment]
Sorry for the straight commit, the propose changes never works for me (what am I doing wrong ☹️? )

File: lib/matplotlib/stackplot.py [hunk 1]
```diff
@@ -21,15 +21,15 @@ def stackplot(axes, x, *args,
 
     Parameters
     ----------
-    x : 1d array of dimension N
+    x : array-like (N)
 
-    y : 2d array (dimension MxN), or sequence of 1d arrays (each dimension 1xN)
+    y : sequence of array-like (N) or 2D array (M, N)
```
Comment (QuLogic) [inline lib/matplotlib/stackplot.py:26]: The lack of a comma and swap of order makes this ambiguous as to whether it's "(sequence of 1D) or 2D", or "sequence of (1D or 2D)".

Comment (tacaswell) [reply lib/matplotlib/stackplot.py:26]:
```suggestion
    y :  2D array (M, N) or sequence of array-like (N)
```

I agree with @QuLogic about the 'or' here being unclear.

Comment (tacaswell) [reply lib/matplotlib/stackplot.py:26]: and it is made extra awkward because y here may or may not be variadic...

**Review by QuLogic — APPROVED** [review]

**Pull request merged**